bqskit/compiler/compile.py

```diff
@@ -669,6 +671,18 @@ def build_workflow(
     if model is None:
         model = MachineModel(input.num_qudits, radixes=input.radixes)
 
+    # Use a registered workflow if model is found in the registry for a given
+    # optimization_level
+    for machine_or_gateset in _workflow_registry:
+        if isinstance(machine_or_gateset, GateSet):
+            gate_set = machine_or_gateset
+        else:
+            gate_set = machine_or_gateset.gate_set
+        gs_match = gate_set == model.gate_set
+        ol_found = optimization_level in _workflow_registry[machine_or_gateset]
+        if gs_match and ol_found:
+            return _workflow_registry[machine_or_gateset][optimization_level]
```

Comment:
I am not sure I follow this check. Why don't you simply check if `(model, optimization_level) in _workflow_registry`?

Reply:
Supporting `GateSet` checking is more valuable than `MachineModel` checking because `GateSet` checking doesn't require knowing a qudit count for the target.